src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,83 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+import { getPostComments, deleteComment } from '../../api/comments';
+import { getPostDetails } from '../../api/posts';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
+export const PostDetails = ({ postId }) => {
+  const [comments, setComments] = useState([]);
+  const [postTitle, setPostTitle] = useState('');
+  const [postDetails, setPostDetails] = useState('');
+  const [commentsHidden, setCommentsHidden] = useState(false);
 
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
+  useEffect(() => {
+    getPostComments(postId)
+      .then(setComments);
 
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
+    getPostDetails(postId)
+      .then((post) => {
+        setPostDetails(post.body);
+        setPostTitle(post.title);
+      });
+  }, [postId, comments]);
```

Comment:
```suggestion
  }, [postId]);
```
забери comments з залежностей, а для того щоб оновлювались дані роби новий запит при нажиманні на кнопки додати і видалити коментар

Reply:
![infinity_loop(comments)](https://user-images.githubusercontent.com/35385361/93324121-21eb5680-f81e-11ea-848a-ba6c15454369.gif)